File: JavaScript/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var S=0;
    var l=0;
    var r = height.length-1;
    while(l<r){
        var h=Math.min(height[l],height[r]);
        S=Math.max(S,h*(r-l)); 
        if(height[l]<height[r]){
            l++;
        }
        else{
            r--;
        }
    }
    return S;
};

File: C/11. Container With Most Water.c
int maxArea(int* height, int heightSize){
    int S=0;
    int l=0;
    int r = heightSize-1;
    while(l<r){
        int h=fmin(height[l],height[r]);
        S=fmax(S,h*(r-l)); 
        if(height[l]<height[r]){
                l++;
        }
        else{
            r--;
        }
    }
    return S;
}
